build: add mips64el support

mutao-loongson opened the pull request:
build: add support for mips64el

V8 now resume supporting for mipsel/mips64el, the support for mips platform 
need to recover.

First, revert commit 9334 e45aa0ed815c2745bcc2bb606d91be64998d.

Second, add linux64-mips64 platform dependent files and update the corresponding 
gypi files and header files for OpenSSL compiling.

<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

refack commented:
Hello @mutao-loongson, and thank you for the contribution!

I would really be happy if we could get some sort of CI for this, preferably under the https://github.com/nodejs/unofficial-builds initiative. Could you help with that?

mutao-loongson commented:
> Hello @mutao-loongson, and thank you for the contribution!
> 
> I would really be happy if we could get some sort of CI for this, preferably under the https://github.com/nodejs/unofficial-builds initiative. Could you help with that?

I would love to do something for nodejs community.  I have read https://github.com/nodejs/unofficial-builds. It only copes with compiling stage of CI, no tests stage. 

What you mean by "some sort of CI"? Could you explain further for that ? What I need to do to offer help? And what need to do to make mips64el to be one of the platforms that nodejs officially supports ? 

Please give me some instructions.

mutao-loongson commented:
OK, I willing to spend time to support node for mipsel-related problems. First, I would following the advice from @refack to response to https://github.com/nodejs/unofficial-builds initiative.  When opportunity matures, then, I would propose adding mipsel machines into the CI matrix. 

Best Regards.

mutao-loongson commented:
Hello, when could this pull request be landed ?

Trott commented:
> Hello, when could this pull request be landed ?

I believe the only remaining requirement is that it pass CI.

CI: https://ci.nodejs.org/job/node-test-pull-request/23913/

Also: /ping @nodejs/build in case anyone there has opinions or wants to do something CI-wise....

Trott commented:
Also: /ping @nodejs/crypto in case anyone wants to look over the OpenSSL stuff

rvagg commented:
This is really sweet. I don't have full confidence in the OpenSSL changes and would love for @sam-github or @shigeki to add their +1 to @bnoordhuis'.
Long-standing policy for platforms not officially supported is something like - we'll accept changes as long as they don't interfere with supported platforms and are not overly intrusive. These changes are very minimal and certainly pass that test IMO. Great work!

mutao-loongson commented:
Yes, I don't change anything about node/openssl on currently supported platform. I just modified some gypi and header files and added auto-generated openssl config output to add mips64el compiling support. But CI can't pass ?!  I am confused! I will setup win2008r2-vs2017 envrionment in virtual machine to try to reproduce the failure and try to figure out what wrong with "node-test-binary-windows-2" test case.

rvagg commented:
I think you just hit a flaky test @mutao-loongson, the re-run is green(ish) so this is good to go I think.

rvagg commented:
Landed in 2a9f1ad...779a243

mutao-loongson commented:
Thanks @rvagg , I would not try to reproduce the failure anymore.
